server/internal/adapter/integration/group.go

```diff
@@ -0,0 +1,234 @@
+package integration
+
+import (
+	"context"
+	"errors"
+
+	"github.com/reearth/reearth-cms/server/internal/adapter"
+	"github.com/reearth/reearth-cms/server/internal/usecase/interfaces"
+	"github.com/reearth/reearth-cms/server/pkg/integrationapi"
+	"github.com/reearth/reearthx/rerror"
+	"github.com/samber/lo"
+)
+
+func (s *Server) GroupFilter(ctx context.Context, request GroupFilterRequestObject) (GroupFilterResponseObject, error) {
+	uc := adapter.Usecases(ctx)
+	op := adapter.Operator(ctx)
+
+	prj, err := uc.Project.FindByIDOrAlias(ctx, request.ProjectIdOrAlias, op)
+	if err != nil {
+		if errors.Is(err, rerror.ErrNotFound) {
+			return GroupFilter404Response{}, err
+		}
+		return GroupFilter500Response{}, err
+	}
+
+	p := fromPagination(request.Params.Page, request.Params.PerPage)
+	gl, pi, err := uc.Group.Filter(ctx, prj.ID(), p, op)
+	if err != nil {
+		if errors.Is(err, rerror.ErrNotFound) {
+			return GroupFilter404Response{}, err
+		}
+		return GroupFilter500Response{}, err
+	}
+	groups := make([]integrationapi.Group, 0, len(gl))
+	for _, g := range gl {
```

Comment:
It is better to implement `ss, err := uc.Schema.FindByGroups(ctx, gIDs, op)` and call it out of the loop to enhance the performance.
and use `lo.Map` to make the code more readable.